odbc/src/api/data.rs

```diff
@@ -1078,6 +1078,30 @@ unsafe fn set_output_binary_helper(
 
 pub mod i16_len {
     use super::*;
+    ///
+    /// set_output_wstring_as_bytes writes [`message`] to the *WChar [`output_ptr`]. [`buffer_len`] is the
+    /// length of the [`output_ptr`] buffer in characters; the message should be truncated
+    /// if it is longer than the buffer length. The number of *BYTES* written to [`output_ptr`]
+    /// should be stored in [`text_length_ptr`].
+    ///
+    /// # Safety
+    /// This writes to multiple raw C-pointers
+    ///
+    pub unsafe fn set_output_wstring_as_bytes(
+        message: &str,
+        output_ptr: Pointer,
+        buffer_len: usize,
+        text_length_ptr: *mut SmallInt,
+    ) -> SqlReturn {
+        let message = message.encode_utf16().collect::<Vec<u16>>();
+        let (len, ret) = set_output_wstring_helper(&message, output_ptr as *mut WChar, buffer_len);
+        // Only copy the length if the pointer is not null
+        if !text_length_ptr.is_null() {
+            *text_length_ptr = (size_of::<WChar>() * len) as SmallInt;
```

Comment:
I think we are better off defining our own `WChar` type that can change to u32 on macos.

Reply:
Then I'll also want to update the tests to multiply by `size_of::<WChar>()`

Reply:
My problem with this approach is that the Wchar size is not OS related but DM related it looks like.
You can use UnixODBC on mac.
I don't know how to have Wchar size match the definition of the DM at runtime. Is this possible?
My general idea was that if we use the template version of the function, we can add a setting in the odbcinst.ini conf to set the encoding/wchar length based on which DM is used, this allows for a runtime definition instead of a compile-time definition.
Or is there something I'm missing?
Also, overall, I don't like WChar because it's a mess. Even UnixODBC has 2 different definition of it. wchar_t (plaform dependent) or unsigned short depending on SQL_WCHART_CONVERT 🙄

Reply:
We are only going to choose one or the other to build on (and it's going to be iODBC since that's what tableau uses on Mac). If someone wants to build our code themselves, that's their problem ;)

Reply:
Also templates are still compile time :(